src/App.js

```diff
@@ -1,5 +1,135 @@
+import { Random, Console } from "@woowacourse/mission-utils";
+import { MESSAGE, ERROR } from "./common/text.js";
+
+const MAX_LENGTH = 5;
+const MIN_LENGTH = 1;
+
 class App {
-  async play() {}
+  async play() {
+    const carArr = await makeCar();
+
+    const move = await inputMoveCount();
+    moveNum(carArr);
+
+    Console.print(`${MESSAGE.GAME_PRINT}`);
+    repeatPrint(move, carArr);
+    finalPrint(carArr);
+  }
 }
+// 1. n개의 자동차 생성
+const makeCar = async () => {
+  let name = "";
+
+  let nameArr = [];
+  try {
+    name = await Console.readLineAsync(`${MESSAGE.GAME_START}`);
+  } catch (error) {
+    return console.error(error.message);
+  }
+  // 유효성 검사
+
+  if (!name) {
+    throw new Error(`${ERROR.NAME_NULL}`);
+  }
+  nameArr = name
+    .trim()
+    .split(",")
+    .map((name) => [name.trim(), ""]);
+
+  let validName = await validationName(nameArr);
+  if (validName) {
+    return nameArr;
+  } else if (validName === "length") {
+    throw new Error(`${ERROR.NAME_LENGTH}`);
+  } else {
+    throw new Error(`${ERROR.NAME_COMMA}`);
+  }
+};
+
+// 2. 자동차 생성 입력값에 대한 유효성 검사
+const validationName = async (nameArr) => {
+  if (nameArr.length === 1) {
+    throw new Error(`${ERROR.NAME_COMMA}`);
+  } else {
+    nameArr.forEach((el, index) => {
+      if (el.length >= `${MAX_LENGTH}` || el.length < `${MIN_LENGTH}`) {
+        throw new Error(
+          `${ERROR.NAME_LENGTH} ${index + 1}번째요소에 대한 에러`
+        );
+      }
+    });
+  }
+  return true;
+};
+
+// 3. 몇번이동할지 입력받기
+const inputMoveCount = async () => {
+  let count = 0;
+  try {
+    count = await Console.readLineAsync(`${MESSAGE.MOVE_COUNT}`);
+  } catch (error) {
+    console.error(error.message);
+  }
+  // 유효성 검사
+  let validCount = await validationCount(count);
+  if (validCount) {
+    return count;
+  } else {
+    throw new Error(`${ERROR.COUNT}`);
+  }
+};
+// 4. 이동 입력값 유효성 검토
+const validationCount = async (count) => {
+  // console.log("이동", count);
+
+  if (isNaN(count)) {
+    throw new Error(`${ERROR.COUNT}`);
+  }
+  return true;
+};
+
+// 5. 이동할 수 있는지 판단하는 번호 생성
+const moveNum = (carArr) => {
+  for (let i = 0; i < carArr.length; i++) {
```

Comment:
에어비앤비 스타일 가이드에서는 i++와 같은 unary operator를 사용하는 것을 권장하지 않고 있습니다.

[13.6](https://github.com/airbnb/javascript#variables--unary-increment-decrement) Avoid using unary increments and decrements (++, --). eslint [no-plusplus](https://eslint.org/docs/rules/no-plusplus)

> Why? Per the eslint documentation, unary increment and decrement statements are subject to automatic semicolon insertion and can cause silent errors with incrementing or decrementing values within an application. It is also more expressive to mutate your values with statements like num += 1 instead of num++ or num ++. Disallowing unary increment and decrement statements also prevents you from pre-incrementing/pre-decrementing values unintentionally which can also cause unexpected behavior in your programs.

해당 내용을 참고해서 ++ 대신 i += 1을 사용해보시는 것도 좋을 것 같아요!